### PR TITLE
WT-12515 Enable Bazel Remote Execution in evergreen scons.py invocation

### PR DESCRIPTION
**Pull request by zackwintermdb**
As part of https://jira.mongodb.org/browse/DEVPROD-257 Bazel is about to be enabled on all platforms by default. This will cause certain targets to be built with Bazel instead of SCons.

Since Bazel uses EngFlow's remote execution system instead of icecream, additional credentials need to be setup to maintain efficient compilation speed.

WT scons.py invocation: https://github.com/10gen/mongo/blob/master/src/third_party/wiredtiger/test/evergreen.yml#L152

The following evergreen function needs to be invoked before the call to SCons above: https://github.com/10gen/mongo/blob/master/etc/evergreen_yml_components/definitions.yml#L178

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -79,6 +79,49 @@ functions:
               export PATH=/opt/mongodbtoolchain/v4/bin:$PATH
             fi
```
Comment (ajmorton) [inline test/evergreen.yml:81]: Can we add a comment explaining these are preparing for the bazel/EngFlow migration? The text from the PR description look perfect for this

File: test/evergreen.yml [hunk 2]
```diff
@@ -79,6 +79,49 @@ functions:
...
+    params:
+      display_name: "generate evergreen engflow bazelrc"
+      shell: bash
+      working_dir: mongo
+      script: |
+        set -o errexit
+        set -o verbose
+
+        # Pulled from evergreen/generate_evergreen_engflow_bazelrc.sh consider
+        # consolidating once prelude.sh is runnable in the perf project.
```
Comment (ajmorton) [inline test/evergreen.yml:109]: Is there a public ticket we can link to for the `prelude.sh` work? If not I'm happy to leave this comment as is.

File: test/evergreen.yml [hunk 3]
```diff
@@ -3841,6 +3884,7 @@ tasks:
           timeout_secs: 86400
       - func: "fetch mongo repo"
       - func: "get project"
+      - func: "get engflow creds"
       - func: "import wiredtiger into mongo"
       - func: "compile mongodb"
```
Comment (ajmorton) [inline test/evergreen.yml:3889]: `compile mongodb` is passing `ICECC=icecc CCACHE=ccache` into the scons call. 
Will this line be changed as part of future DEVPROD-257 work, or should the WiredTiger team keep an eye on this and update it ourselves?

**Review by ajmorton — COMMENTED** [review]
Thanks for this!
I have a question or two about future work, but the changes are looking good 👍

File: test/evergreen.yml (hunk 2 above)
Comment (zackwintermdb) [reply test/evergreen.yml]: I don't think we had one before but I created https://jira.mongodb.org/browse/SERVER-86966 to track it

File: test/evergreen.yml (hunk 1 above)
Comment (zackwintermdb) [reply test/evergreen.yml]: Done

**Review by ajmorton — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**